code/switcher.py

```diff
@@ -68,10 +68,19 @@ def get_words(name):
 class Actions:
     def switcher_focus(name: str):
         """Focus a new application by  name"""
+        running = ctx.lists["self.running"]
+        wanted_app = None
+        for running_name in running.keys():
+            if running_name == name or running_name.lower().startswith(name):
+                wanted_app = running[running_name]
+                break
+        if wanted_app is None:
+            return
+
         for app in ui.apps():
-            # print(f"--------- app.name:{app.name}  app.bundler:{app.bundle}")
-            if name in app.name and not app.background:
-                app.focus()
+            if app.name == wanted_app and not app.background:
+                os.system("i3-msg '[class=\"(?)%s\"] focus'" % app.name)
```

Comment:
This looks like a i3-specific call - i assume there an issue with app.focus() for i3?

Reply:
oh right good catch, app.focus can be used in general . I started to use monkey patching for that kind of stuff so these don't show up in diffs, but forgot to switch this part back

Reply:
I switched this back and also fixed a couple other warnings I noticed